icelake/tests/delete_test.rs

```diff
@@ -257,25 +257,104 @@ impl PositionDeleteTest {
         )
     }
 
-    async fn run(mut self) {
+    async fn run_position_delete_test(mut self) {
         self.init();
         self.write_data_with_icelake().await;
-        self.delete_data().await;
+        self.position_delete_data().await;
     }
 
-    pub fn block_run(self) {
+    async fn equality_delete_data(&mut self) {
+        let mut table = self.create_icelake_table().await;
+        log::info!(
+            "Real path of table is: {}",
+            table.current_table_metadata().location
+        );
+
+        let mut writer = table
+            .writer_builder()
+            .await
+            .unwrap()
+            .build_equality_delete_writer(vec![1, 2])
+            .await
+            .unwrap();
+
+        let records = RecordBatch::try_new(
+            Arc::new(
+                table
+                    .current_table_metadata()
+                    .current_schema()
+                    .unwrap()
+                    .clone()
+                    .try_into()
+                    .unwrap(),
+            ),
+            vec![
+                Arc::new(Int64Array::from(vec![1])),
+                Arc::new(Int64Array::from(vec![1])),
+            ],
+        )
+        .unwrap();
+
+        writer.write(records).await.unwrap();
+
+        let result = writer.close().await.unwrap().pop().unwrap();
+
+        // Commit table transaction
+        {
+            let mut tx = Transaction::new(&mut table);
+            tx.append_delete_file(vec![result
+                .with_equality_ids(vec![1, 2])
+                .with_partition_value(self.partition_value.clone().unwrap())
+                .build()]);
+            tx.commit().await.unwrap();
+        }
+        self.poetry.run_file(
+            "execute.py",
+            [
+                "-s",
+                &self.spark_connect_url(),
+                "-q",
+                "delete from s1.t2 where vlong = 1;",
```

Comment:
```suggestion
                "delete from s1.t2 where id=1 and vlong = 1;",
```
Though the results are the same, I still prefer to make this change to ensure the semantics are same.